src/common.cpp

```diff
@@ -19,29 +19,27 @@ NAMESPACE_BEGIN(detail)
     [[noreturn]]
 #endif
 void raise(const char *fmt, ...) {
-    char buf[512], *ptr = buf;
+    char buf[512];
     va_list args;
 
     va_start(args, fmt);
-    size_t size = vsnprintf(ptr, sizeof(buf), fmt, args);
+    size_t size = vsnprintf(buf, sizeof(buf), fmt, args);
     va_end(args);
 
     if (size < sizeof(buf))
         throw std::runtime_error(buf);
 
-    ptr = (char *) malloc(size + 1);
-    if (!ptr) {
+    auto allocated = std::make_unique<char[]>(size + 1);
```

Comment:
This creates a dependence on `operator new`. My plan for the `libnanobind` library was to only use standard C memory allocation, and to eventually route this through `PyMalloc`.